lib/utils/ProductSearchQueryConfiguration.dart

```diff
@@ -5,21 +5,40 @@ import 'package:openfoodfacts/utils/ProductFields.dart';
 
 /// Product Search Query Configuration, that helps building search API URI
 class ProductSearchQueryConfiguration {
-  OpenFoodFactsLanguage? language;
+  List<OpenFoodFactsLanguage> languages;
   // Allow apps to directly provide the language code and country code without
   // having to use the OpenFoodFactsLanguage helper.
   String? lc;
   String? cc;
   List<ProductField>? fields;
   List<Parameter> parametersList;
 
-  ProductSearchQueryConfiguration({
-    this.language,
-    this.lc,
-    this.cc,
-    this.fields,
-    required this.parametersList,
-  });
+  @deprecated
+  OpenFoodFactsLanguage? get language =>
+      languages.isNotEmpty ? languages.first : null;
+  @deprecated
+  set language(OpenFoodFactsLanguage? value) {
+    languages.clear();
+    if (value != null) {
+      languages.add(value);
+    }
+  }
+
+  ProductSearchQueryConfiguration(
+      {this.languages = const [],
+      @deprecated OpenFoodFactsLanguage? language,
+      this.lc,
+      this.cc,
+      this.fields,
+      required this.parametersList}) {
+    if (languages.isNotEmpty && language != null) {
+      throw ArgumentError(
+          'Please provide either language or languages, not both');
+    }
+    if (language != null) {
+      languages = [language];
```

Comment:
Things are getting even more confused between `lc`, `languages` and `language`.

Reply:
You're right, it's becoming more confusing.

But while the `Product.*translated` fields were added by me not very long time ago and I felt confident enough to replace the old ones with the new ones because it's likely used by my project only, the `*Configuration.language` field was added long before that, so the field probably is used by a lot of people.

Some solutions I see:
- completely remove the `language` field, make the `languages` field to be the only one.
- return the old form to the `language` field, replace `languages` field with `translationsLanguages` field, which would be used only for the `Product.*translated` fields (the old `language` field will be used to set `'lc'` as before).

What do you think?
Maybe there's a better way I don't see?

Reply:
Keeping `lc`, `language` and `languages` is a mess.
I guess we could deprecate both `language` (as you suggested) and `lc`.